pkg/expr/classic/classic.go

```diff
@@ -11,52 +11,44 @@ import (
 	"github.com/grafana/grafana/pkg/expr/mathexp"
 )
 
-// ConditionsCmd is command for the classic conditions
-// expression operation.
-type ConditionsCmd struct {
+// ClassicConditionsCmd is a command that supports the reduction and comparison of conditions.
+//
+// A condition in ClassicConditionsCmd can reduce a time series; or contain either an instant metric
+// or the result of another expression; and check if it exceeds a threshold, falls within a
+// range, or does not contain a value.
+//
+// If ClassicConditionsCmd contains more than one condition, it reduces the boolean outcomes of the
+// threshold, range or value checks using the logical operator of the right hand side condition
```

Comment:
So our operators don't have precedence or grouping? Hrm. Good explanation though.

Reply:
No precedence, it iterates over the conditions and either AND or ORs the current condition with a local variable that stores the boolean outcome.